### PR TITLE
WIP: vector-based support using CanvasRenderingContext2D

### PR DESCRIPTION
**Pull request by hfmanson**
This PR adds vector-based support using CanvasRenderingContext2D
- I remove SVG support for now
- Description in README.md

TODO
- error handling on non existing vector types, currently only 'canvas' is supported
- Codemirror support and error handling

### CODE REVIEW

File: src/js/graphics.js [hunk 1]
```diff
@@ -29,6 +29,26 @@ function createTextSprite(name, text, colors, scale) {
     return canvas;
 }
 
+// Create and return a custom instructions sprite canvas
+function createJsonSprite(name, vector) {
+    var canvas = makeSpriteCanvas(name);
+    var context = canvas.getContext('2d');
+    if (vector.w) canvas.width *= vector.w;
+    if (vector.h) canvas.height *= vector.h;
+    var json = JSON.parse("[" + vector.data + "]");
```
Comment (hfmanson) [inline src/js/graphics.js:38]: JSON.parse throws a SyntaxError exception when it can't parse the input string

File: src/js/graphics.js [hunk 2]
```diff
@@ -100,12 +120,18 @@ function regenText() {
 var editor_s_grille=[[0,1,1,1,0],[1,0,0,0,0],[0,1,1,1,0],[0,0,0,0,1],[0,1,1,1,0]];
 
 var spriteImages;
+
+function handleVector(name, vector) {
+    return vector.type === 'canvas' ? createJsonSprite(name, vector)
+    : undefined;
+}
+
```
Comment (hfmanson) [inline src/js/graphics.js:128]: TODO: currently `undefined` is returned when type does not equal `canvas`, should probably done another way, e.g. by throwing an exception

Comment (david-pfx) [reply src/js/graphics.js:128]: The logic is OK, but I prefer to return null. I use exceptions only for things I can't and shouldn't happen.

File: src/js/graphics.js (hunk 1 above)
Comment (david-pfx) [reply src/js/graphics.js:38]: So wrap it in a try-catch block, return null.

**Pull request merged**